platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/mappingToJvm/RuntimeFunctionResolver.kt

```diff
@@ -44,44 +47,33 @@ class MemberFunctionResolver(private val configureLambdaHandler: ConfigureLambda
         val parameterBindingStub = schemaFunction.parameters.associateWith { Any() }
         val hasConfigureLambda = (schemaFunction.semantics as? FunctionSemantics.ConfigureSemantics)?.configureBlockRequirement?.allows ?: false
 
+        fun signature(function: KFunction<*>): String =
```

Comment:
💭 I see why non-public API is used here, and while IMO quite undesirable, it might work for us given that the Kotlin version in the Gradle runtime is under our control.

Consider, however, a different internal API to use here: `KCallableImpl.descriptor` has `overriddenDescriptors` which seem to be usable to match a function with its super declaration.